server/src/main/java/org/elasticsearch/action/admin/indices/shrink/ResizeRequest.java

```diff
@@ -80,6 +80,9 @@ public ActionRequestValidationException validate() {
         if (type == ResizeType.SPLIT && IndexMetaData.INDEX_NUMBER_OF_SHARDS_SETTING.exists(targetIndexRequest.settings()) == false) {
             validationException = addValidationError("index.number_of_shards is required for split operations", validationException);
         }
+        if (copySettings != null && copySettings == false) {
+            validationException = addValidationError("[copySettings] can not be explicitly set to [false]", validationException);
+        }
```

Comment:
Should we instead do this validation on the setter method? That way:
1. The stack trace that is thrown will contain the call stack that actually tried to set `copySettings` in an invalid way making it easier to debug
2. Allows us to put the validation only on the setter method (and have the rest handler call the setter method) so we ensure both the rest and transport validation are consistent and fail early.